### PR TITLE
Fix bug: correct the output shape of  `aten::index.Tensor`

### LINKED ISSUE #1274: 🐛 [Bug] Advanced Indexing/GatherND compilation causes error: `isObject() INTERNAL ASSERT FAILED at "libtorch/_virtual_includes/ATen/ATen/core/ivalue_inl.h":123, please report a bug to PyTorch. Expected Object but got None`
##  Bug Description
PyTorch uses advanced indexing to implement GatherND. For example:
```
torch[[0,1], :, None, torch.tensor((0,1))]
```
is considered a valid indexing operation.
However, when applying TensorRT compilation, we see the following error:
```
isObject() INTERNAL ASSERT FAILED at "bazel-out/k8-opt/bin/external/libtorch/_virtual_includes/ATen/ATen/core/ivalue_inl.h":123, please report a bug to PyTorch. Expected Object but got None
```

## To Reproduce
Run with `nvcr.io/nvidia/pytorch:22.07-py3` container (`gathernd.py`):
```
  import tensorrt
  import torch
  import torch_tensorrt

  class GatherNDListModel(torch.nn.Module):
      def __init__(self):
          super().__init__()

      def forward(self, x):
          return x[[1,0], :, [1,0]]


  class GatherNDTensorModel(torch.nn.Module):
      def __init__(self):
          super().__init__()
          self.index = torch.tensor((1,0))

      def forward(self, x):
          return x[self.index, :, self.index]

  if __name__ == '__main__':
      torch.manual_seed(0)
      t = torch.randn(3, 4, 5).cuda()

      try:
          m = GatherNDListModel()
          o = m(t)
          print(f"PyTorch: {o.shape}")

          mt = torch_tensorrt.compile(m, inputs=[t], truncate_long_and_double=True)
          o = mt(t)
          print(f"TRT: {o.shape}")
      except Exception as err:
          print(f"GatherND list indexing failed: {err}")

      try:
          m = GatherNDTensorModel()
          o = m(t)
          print(f"PyTorch: {o.shape}")

          mt = torch_tensorrt.compile(m, inputs=[t], truncate_long_and_double=True)
          o = mt(t)
          print(f"TRT: {o.shape}")
      except Exception as err:
          print(f"GatherND tensor indexing failed: {err}")
```
Outputs the following:
```
PyTorch: torch.Size([2, 4])
WARNING: [Torch-TensorRT] - Truncating weight (constant in the graph) from Int64 to Int32
GatherND list indexing failed: isObject() INTERNAL ASSERT FAILED at "bazel-out/k8-opt/bin/external/libtorch/_virtual_includes/ATen/ATen/core/ivalue_inl.h":123, please report a bug to PyTorch. Expected Object but got None
PyTorch: torch.Size([2, 4])
WARNING: [Torch-TensorRT] - Truncating weight (constant in the graph) from Int64 to Int32
GatherND tensor indexing failed: isObject() INTERNAL ASSERT FAILED at "bazel-out/k8-opt/bin/external/libtorch/_virtual_includes/ATen/ATen/core/ivalue_inl.h":123, please report a bug to PyTorch. Expected Object but got None
```
Note that the pytorch version (ie. no-TRT) model outputs the expected shapes, but the TRT conversion fails. 

## Expected behavior
The TRT compilation should succeed and output the same output as the non-TRT version.

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages
`nvcr.io/nvidia/pytorch:22.07-py3` 

## Additional context
https://partners.nvidia.com/Bug/ViewBug/3735309

### PR DESCRIPTION
**Pull request by ruoqianguo**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

When i run the below code, i found the shape of trt output mismatched pytorch. Thanks @peri044 for helping me find this bug.
```
import torch
import torch.nn as nn
import torch_tensorrt

class IndexTest(nn.Module):
    def __init__(self):
        super(IndexTest, self).__init__()

    def forward(self, input: torch.Tensor, x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
        return input[x.long(), y.long(), z.long(), :]

def reproduce_error():
    torch_tensorrt.logging.set_reportable_log_level(torch_tensorrt.logging.Level.Graph)
    model = IndexTest()

    input = torch.randn([4, 8, 8, 4]).cuda()
    x = torch.full([4, 13, 1], 1).int().cuda()
    y = torch.full([4, 13, 1], 2).int().cuda()
    z = torch.full([4, 13, 1], 3).int().cuda()
    test_output = model.forward(input, x, y , z)
    traced_model = torch.jit.trace(model, [input, x, y, z])

    trt_model = torch_tensorrt.compile(traced_model.eval().cuda(), inputs=[input, x, y, z],  **{
            "truncate_long_and_double": True,
        })
    converted_output = trt_model.forward(input, x, y ,z)

    print(test_output.shape)
    print(converted_output.shape)
    print(torch.sum(test_output-converted_output))

reproduce_error()
```

Fixes #1274 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by narendasan — APPROVED** [review]
LGTM
